### PR TITLE
Add rest api ingress

### PR DESCRIPTION
**Pull request by luigibk**
### JIRA link (if applicable) ###

[AB#244]

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: templates/controller-service.yaml [hunk 1]
```diff
@@ -9,6 +9,9 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
+{{- if .Values.controller.ingress.enabled }}
+  type: ClusterIP
```
Comment (timja) [inline templates/controller-service.yaml:13]: should this be configurable?

File: templates/ingress.yaml [hunk 2]
```diff
@@ -30,3 +30,36 @@ spec:
           serviceName: neuvector-service-webui
           servicePort: 8443
 {{- end }}
+---
+{{- if and .Values.controller.enabled .Values.controller.ingress.enabled -}}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: neuvector-webui-ingress
```
Comment (timja) [inline templates/ingress.yaml]: is webui the right name?

File: templates/controller-service.yaml (hunk 1 above)
Comment (luigibk) [reply templates/controller-service.yaml:13]: I don't think so. The ingress is pointing to it and that's how it's exposed. Using NodePort this gets also exposed in a different way and using LoadBalancer exposes it also in a different way I think, but it sounds very confusing. ClusterIP exposes the service only on a cluster internal ip and the ingress does the external access bit. Unless I am missing something... :-)

File: templates/ingress.yaml (hunk 2 above)
Comment (luigibk) [reply templates/ingress.yaml]: Fixed. Thanks!

File: templates/controller-service.yaml (hunk 1 above)
Comment (timja) [reply templates/controller-service.yaml:13]: sure, it's a thing that is sometimes made configurable to make it more flexible in other environments but let's see what the neuvector people say

File: templates/ingress.yaml [hunk 3]
```diff
@@ -30,3 +30,36 @@ spec:
...
+    secretName: {{ .Values.controller.ingress.secretName }}
+{{- end }}
+{{- end }}
+  rules:
+  - host: {{ .Values.controller.ingress.host }}
+    http:
+      paths:
+      - path: {{ .Values.controller.ingress.path }}
+        backend:
+          serviceName: neuvector-service-restapi
```
Comment (timja) [inline templates/ingress.yaml]: am I missing something? I can't see the service for this

File: templates/ingress.yaml [hunk 4]
```diff
@@ -30,3 +30,36 @@ spec:
...
+          servicePort: 10443
```
Comment (timja) [inline templates/ingress.yaml:64]: also can't see this port

File: templates/ingress.yaml (hunk 3 above)
Comment (luigibk) [reply templates/ingress.yaml]: That is in controller-service.yaml. https://github.com/hmcts/neuvector-helm/pull/1/files#diff-dace72095c9160121bd33265cd9dfafa

File: templates/ingress.yaml (hunk 4 above)
Comment (luigibk) [reply templates/ingress.yaml:64]: That is in controller-service.yaml. https://github.com/hmcts/neuvector-helm/pull/1/files#diff-dace72095c9160121bd33265cd9dfafa

Comment (timja) [reply templates/ingress.yaml:64]: where?

**Review by timja — APPROVED** [review]

**Pull request merged**